src/lib/job.ml

```diff
@@ -0,0 +1,127 @@
+open Internal_pervasives
+
+module Specification = struct
+  type t =
+    | Kube of Kube_job.Specification.t
+    | Local_docker of Local_docker_job.Specification.t
+  [@@deriving yojson, show]
+  let kind =
```

Comment:
It's just a way of pattern matching on the kind of Job without exposing the contents of `t`.
I ended up exposing `t` as a `private` type anyway (I think mostly for the command **display** in the Ketrew plugin) but the goal would be to make it completely abstract at some point.